Tests/Listener/RestListenerLocaleTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Paysera\Bundle\RestBundle\Tests;
+
+use Mockery;
+use Psr\Log\NullLogger;
+use Mockery\MockInterface;
+use PHPUnit_Framework_TestCase;
+use Paysera\Bundle\RestBundle\ApiManager;
+use Symfony\Component\HttpFoundation\Request;
+use Paysera\Bundle\RestBundle\Service\RequestLogger;
+use Paysera\Bundle\RestBundle\Listener\RestListener;
+use Paysera\Bundle\RestBundle\Service\ExceptionLogger;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Paysera\Bundle\RestBundle\Service\ParameterToEntityMapBuilder;
+use Paysera\Component\Serializer\Factory\ContextAwareNormalizerFactory;
+
+class RestListenerLocaleTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var MockInterface|GetResponseEvent
+     */
+    private $responseEvent;
+
+    public function setUp()
+    {
+        $this->responseEvent = Mockery::mock(GetResponseEvent::class);
+    }
+
+    /**
+     * @param array $acceptedLocales
+     * @param string $expected
+     * @param string|null $acceptLanguage
+     *
+     * @dataProvider dataProviderForTestLocaleIsBeingSet
+     */
+    public function testCorrectLocaleIsBeingSet(array $acceptedLocales, $expected, $acceptLanguage = null)
+    {
+        $restListener = $this->createRestListener($acceptedLocales);
+
+        $request = new Request();
```

Comment:
We should set some default locale to see that it's not changed if corresponding headers are not passed